### DISCUSSION TITLE
Golang  occured an error

### DISCUSSION THREAD

**Discussion by longintlong** [ayamir/nvimdots#1109] [Q&A] [answered]
Hi everyone, i met an error in a golang project like the folloing pic.
<img width="506" alt="image" src="https://github.com/ayamir/nvimdots/assets/39492651/9d428653-2b77-471a-aa4f-c4d9475f1447">

my env is fedora 38, neovim is 0.9, i have installed the golang-ci-lint and gopls

Thx!

**Comment by charliie-dev** [comment]
it doesn't looks like a nvim config issue, report to `golang-ci-lint` and `gopls` might give u more help.

**Reply by longintlong** [reply]
thank u, and i type `:scriptnames` to try finding the source of error, the result is
~/.local/share/nvim/site/lazy/vim-go/autoload/go/lsp.vim

**Reply by longintlong** [reply]
oh~~ i found the latest [PR](https://github.com/fatih/vim-go/pull/3616) of vim-go, this patch seems to use un-standard vim function `state()`

```
function! s:debugasync(timer) abort
  let s:logtimer = 0

  " set the timer to try again if Vim is in a state where we don't want to
  " change the window.
  let l:state = state('a')
  let l:mode = mode(1)
  if len(l:state) > 0 || l:mode[0] == 'v' || l:mode[0] == 'V' || l:mode[0] == 's' || l:mode =~ 'CTRL-V'
    let s:logtimer = timer_start(go#config#DebugLogDelay(), function('s:debugasync', []))
    return
  endif
```

i'm not sure if we need to define the function ourselves

**Comment by longintlong** [comment, marked as answer]
the [fix](https://github.com/ayamir/nvimdots/commit/7df41c45cb0c58b59d6ab994b4022b901e4ab832) was already merged in main. Now vim-go is update to the latest and the error was fixed.
